feat: Parameterized linked service for Azure SQL pipelines (#46)
delphix/dcs-for-azure-templates: dcsazure_AzureSQL_to_AzureSQL_discovery_pl/README.md

### Prerequisites
1. Configure the hosted metadata database and associated Azure SQL service (version `V2025.01.15.0`+).
1. Configure the DCS for Azure REST service.
1. Configure the AzureSQL linked service. 
   * It is helpful for the linked service to be parameterized with the following parameter:

>       * `LS_DATABASE` - database name in the linked service

Author: I've kept it in the same format as it is present in Snowflake.

> 1. Configure the AzureSQL linked service. 
>    * It is helpful for the linked service to be parameterized with the following parameter:
>       * `LS_DATABASE` - database name in the linked service

Reviewer: I was going to suggest the following (like in the other [PR for Azure SQL MI](https://github.com/delphix/dcs-for-azure-templates/pull/47#discussion_r1998693335)) but then I saw this thread. Feel free to ignore it if its not helpful:

```suggestion
1. Configure the Azure SQL MI linked service.
   * Parameterize the linked service with the following parameter:
      * `LS_DATABASE` - The name of the database to connect to
```

>       * `LS_DATABASE` - database name in the linked service

Author: keeping it in the same format as it is present in Snowflake
